static/app/views/issueDetails/groupTagValues.tsx

```diff
@@ -43,55 +37,109 @@ type RouteParams = {
 type Props = {
   baseUrl: string;
   group: Group;
-  organization: Organization;
   environments?: string[];
   project?: Project;
-} & RouteComponentProps<RouteParams, {}>;
-
-type State = {
-  tag: Tag | null;
-  tagValueList: TagValue[] | null;
-  tagValueListPageLinks: string;
 };
 
 const DEFAULT_SORT = 'count';
 
-class GroupTagValues extends DeprecatedAsyncComponent<
-  Props & DeprecatedAsyncComponent['props'] & WithRouterProps,
-  State & DeprecatedAsyncComponent['state']
-> {
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {environments: environment, organization} = this.props;
-    const {groupId, tagKey} = this.props.params;
-    return [
-      ['tag', `/organizations/${organization.slug}/issues/${groupId}/tags/${tagKey}/`],
-      [
-        'tagValueList',
-        `/organizations/${organization.slug}/issues/${groupId}/tags/${tagKey}/values/`,
-        {query: {environment, sort: this.getSort()}},
-      ],
-    ];
-  }
-
-  getSort(): string {
-    return this.props.location.query.sort || DEFAULT_SORT;
-  }
+function useTagQueries({
+  group,
+  tagKey,
+  environments,
+  sort,
+}: {
+  group: Group;
+  sort: string | string[];
+  environments?: string[];
+  tagKey?: string;
+}) {
+  const organization = useOrganization();
+
+  const {
+    data: tagValueList,
+    isLoading: tagValueListIsLoading,
+    isError: tagValueListIsError,
+    getResponseHeader,
+  } = useApiQuery<TagValue[]>(
+    [
+      `/organizations/${organization.slug}/issues/${group.id}/tags/${tagKey}/values/`,
+      {query: {environment: environments, sort}},
+    ],
+    {staleTime: 0, retry: false}
+  );
+  const {data: tag, isError: tagIsError} = useApiQuery<Tag>(
+    [`/organizations/${organization.slug}/issues/${group.id}/tags/${tagKey}/`],
+    {staleTime: 0, retry: false}
+  );
+
+  useEffect(() => {
+    if (tagIsError) {
+      addMessage(t('Failed to fetch total tag values'), 'error');
```

Comment:
What about using the onError option in rq here?

Reply:
`onError` and the other callbacks have been removed in the new version: https://github.com/TanStack/query/discussions/5279

But good point that the useEffect isn't great (especially for reusuable query hooks). What we _should_ do is add a new option to `useApiQuery` like `toastErrorMessage` and add the logic to the global query options. That way we won't need to write a useEffect and we can't have duplicate error messages

Reply:
@malwilley ahhh, I did not know it's getting removed. 

My only concern is that at some point, someone will likely say "i don't want the toast error message" or they'll want to change something related to react query (like staletime or retry) and the component will eventually get littered with one off changes and differently named props. There is a similar risk with returning only a part of the react query object which means that we will inevitably end up with different shapes of return types on these data fetching hooks, which is long term not a good outlook for the readability and consistency of the codebase. e.g someone may return loading vs isLoading or error vs isError etc. which quickly becomes confusing and annoying on a daily basis.

Regardless, I don't think this PR is a good place to decide this, getting rid of AsyncView is the priority and we can see how/if we want to reuse these in the future.